src/models/User.test.ts

```diff
@@ -0,0 +1,72 @@
+import { connect, disconnect } from '../connectToMongo'
+import { User } from './User'
+import { IProfile } from './User.d'
+
+// testing cluser of mongodb, to replace with docker service
+const MONGO_DB_HOST = 'cluster0.xzpkf.mongodb.net',
+	MONGO_DB_NAME = "test",
+	MONGO_DB_USERNAME = "dbUser",
+	MONGO_DB_PASSWORD = 'EllxyUiddWRqappW'
+
+const connstring: string = `mongodb+srv://${MONGO_DB_USERNAME}:${MONGO_DB_PASSWORD
+	}@${MONGO_DB_HOST
+	}/${MONGO_DB_NAME}?retryWrites=true&w=majority`
+
+describe("User model", () => {
+	beforeAll(async () => {
+		await connect(connstring)
+	})
+
+	afterAll(async () => {
+		await disconnect()
+	})
+
+	describe("static method .findOneOrCreate works properly when", () => {
+		const mockedToken = 'test-token-hash-123'
+		const mockedProfile: IProfile = {
+			id: 'test-id-123',
+			displayName: 'test-user-name'
+		}
+
+		let uuidToPassBetweenTests: string = ""
```

Comment:
Correct me if im wrong, but shouldn't tests be executed in separate context? Should not be depended on other tests?

Reply:
i build separate context - with inner describe. If this would be unit tests, you are right. but here in intergration test (test connect to db, doing stuff and disconnect) you can build logic like this, for example to remove added document during afterAll cleaning phase